notebook/autogen_agent_MathChat.ipynb

```diff
@@ -0,0 +1,752 @@
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The config list looks like the following:\n",
+    "```python\n",
+    "config_list = [\n",
+    "    {\n",
+    "        'model': 'gpt-4',\n",
+    "        'api_key': '<your OpenAI API key here>',\n",
+    "    },  # only if OpenAI API key is found\n",
+    "    {\n",
+    "        'model': 'gpt-4',\n",
+    "        'api_key': '<your first Azure OpenAI API key here>',\n",
+    "        'api_base': '<your first Azure OpenAI API base here>',\n",
+    "        'api_type': 'azure',\n",
+    "        'api_version': '2023-03-15-preview',\n",
+    "    },  # only if the at least one Azure OpenAI API key is found\n",
+    "    {\n",
+    "        'model': 'gpt-4',\n",
+    "        'api_key': '<your second Azure OpenAI API key here>',\n",
+    "        'api_base': '<your second Azure OpenAI API base here>',\n",
+    "        'api_type': 'azure',\n",
+    "        'api_version': '2023-03-15-preview',\n",
+    "    },  # only if the second Azure OpenAI API key is found\n",
+    "]\n",
+    "```\n",
+    "\n",
+    "You can directly override it if the above function returns an empty list, i.e., it doesn't find the keys in the specified locations."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Construct agents for MathChat\n",
+    "\n",
+    "We start by initialzing the `AssistantAgent` and `MathUserProxyAgent`. The system message needs to be set to \"You are a helpful assistant.\" for MathChat. The detailed instructions are given in the user message. Later we will use the `MathUserProxyAgent.generate_prompt` to combine the instructions and a math problem for an intial prompt to be sent to the LLM assistant."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from flaml.autogen.agent.assistant_agent import AssistantAgent\n",
+    "from flaml.autogen.agent.math_user_proxy_agent import MathUserProxyAgent\n",
+    "\n",
+    "oai.ChatCompletion.start_logging()\n",
+    "\n",
+    "# 1. create an assistant which is essentially a AssistantAgent instance named \"coding_agent\"\n",
+    "assistant = AssistantAgent(\"assistant\", \n",
+    "                           system_message=\"You are a helpful assistant.\",\n",
+    "                           request_timeout=600, \n",
+    "                           seed=42, \n",
+    "                           config_list=config_list)\n",
+    "\n",
+    "# 2. create the MathUserProxyAgent instance named \"mathproxyagent\"\n",
+    "# By default, the human_input_mode is \"NEVER\", which means the agent will not ask for human input.\n",
+    "mathproxyagent = MathUserProxyAgent(name=\"MathChatAgent\", \n",
+    "                                    human_input_mode=\"NEVER\",\n",
+    "                                    use_docker=False)"
```

Comment:
format

Reply:
where is the `receive` call?

Reply:
The receive call is in the examples. Here only the two agents are defined.